### DISCUSSION TITLE
MiSTer CIFS - Concept of multiple collections

### DISCUSSION THREAD

**Discussion by jamescadd** [retronas/discussions#68] [Ideas]
Hello, for many retro consoles I have multiple "collections" of games. Some are complete NoIntro sets, some are sets of translated ROMS, and others are "best of" collections I've found over the years. MAME is "notorious" for this as there are several versions of that romset considered definitive.

Would it be possible for RetroNAS to help swap out the MiSTer share, or parts of it (per console), depending on the library I choose? The use case would be sitting down for a couch gaming session, deciding I want to check out some RPG translations, choose that collection in RetroNAS easily from cockpit (on mobile) and then turning on MiSTer. I have MiSTer setup to wait for that network share on boot.

Any thoughts on whether this is doable, or if others want something similar? Fwiw I have yet to investigate ROM library management software or the SmokeMonster feature in RetroNAS today.

**Comment by sairuk** [comment]
from a technical perspective it's doable, a cockpit module would probably just exec a script on the system to return a list of collections (as directories) and then manage accessibility through symlinks from whatever folder. If that was the solution I don't know why you'd do this and not just have collection subfolders in your MiSTer share though.

We've already been discussing how to manage 'multi-sets' like TOSEC or MAMESL which we don't really have a good solution for at this stage. I used to do symlinks to files based on regex patterns to create collections then point my stuff specifically to the folder with the symlinks but that's not feasible atm

entirely going to be a per-person want tho so making it a _thing_ would need some serious community need/input.

**Reply by jamescadd** [reply]
For some reason it never occurred to me that I could have subfolders within each console for the various collections. Problem solved, thanks!

**Reply by sairuk** [reply]
yeh nps, this is exactly why the discussions are useful imo.
